esp/services/ws_sql/SQL2ECL/HPCCSQLTreeWalker.cpp

```diff
@@ -304,7 +304,12 @@ ISQLExpression * HPCCSQLTreeWalker::expressionTreeWalker(pANTLR3_BASE_TREE exprA
             case LIKE_SYM:
             case NOT_LIKE:
                 leftexp.set(expressionTreeWalker((pANTLR3_BASE_TREE)(exprAST->getChild(exprAST, 0)),exprAST));
+                if (leftexp->getExpType() != Value_ExpressionType && strnicmp(leftexp->getECLType(),"DATASET",7)==0)
```

Comment:
similar question to the above should we be using regex instead of strnicmp here? Is there a scenario where we might get whitespace? \s*DATASET\s*/i

Reply:
that ecltype is set internally, we should be fine with the string-wise conditional as is (although we definitely should use codes instead).